### PR TITLE
Add a shutdown and restart hook

### PR DESCRIPTION
**Pull request by tamis-laan**
Getting the hang of this, please review.

### CODE REVIEW

**Comment by ramnes** [comment]
I think that intuitively I would have expected a "restart" hook to happen after a restart, not before. @tych0 what do you think? If you're good with the current naming, feel free to merge.

**Comment by tamis-laan** [comment]
That does not sound very intuitive to me, usually you want to do some work on startup and then do some work on shutdown/restart. 

If the restart hook was triggered after the restart it would be triggered at the same time as the startup hook!?

**Comment by m-col** [comment]
What about "before_restart"?

File: libqtile/hook.py [hunk 1]
```diff
@@ -87,6 +87,24 @@ def startup_complete(self, func):
         """
         return self._subscribe("startup_complete", func)
 
+    def shutdown(self, func):
+        """Called when qtile is shutdown
```
Comment (tych0) [inline libqtile/hook.py:91]: Maybe change "when" to "before"? That clarifies most of the discussion r.e. names, I think.

**Comment by ramnes** [comment]
Alright let's gooooo

**Pull request merged**